PYME/LMVis/visCore.py

```diff
@@ -672,21 +672,40 @@ def _populate_open_args(self, filename):
             from scipy.io import loadmat
         
             mf = loadmat(filename)
-            if not 'x' in mf.keys():
```

Comment:
```python
if (len([k for k in mf.keys() if not k.startswith('_')]) < 3) and not 'x' in mf.keys():
    # all the data is probably packed in a single variable
    dlg = importTextDialog.ImportMatDialog(self, [k for k in mf.keys() if not k.startswith('__')])
    ret = dlg.ShowModal()
   # rest of single var logic ....
elif not 'x' in mf.keys() and  not 'x' in mf.keys():
    importMatlabDialog(mf)
    # rest of key mapping logic
```

Reply:
Adding the `len([k for k in mf.keys() if not k.startswith('_')]) < 3)` is necessary. What's the argument for changing the nested if structure to an `elif`? The nested structure allows us to do a single check for the presence of `x` instead of two checks, and I think makes a more readable decision tree for the next person to come look at this section of code. I've commented what each conditional statement is "thinking".

Reply:
Also, I think `'x' not in mf.keys()` is clearer than `not 'x' in mf.keys()`, but you seem to have a preference for the latter. I assumed it was a legacy preference, but you reiterated here. Is there a reason for this style? Should I switch to using this style?

Reply:
I guess the nested vs. flat comes down to code style. In general, I agree a flat `if` statement is better than a nested `if` statement. However, I think a nested `if` statement is better than an `elif` statement without an `else`. Dangling `elif`s bother me, but that's just a personal preference. If you feel strongly, we can change it.